crates/oxc_semantic/src/binder.rs

```diff
@@ -439,6 +439,30 @@ impl<'a> Binder<'a> for TSModuleDeclaration<'a> {
 
 impl<'a> Binder<'a> for TSTypeParameter<'a> {
     fn bind(&self, builder: &mut SemanticBuilder) {
+        if matches!(
+            builder.nodes.parent_kind(builder.current_node_id),
+            Some(AstKind::TSInferType(_))
+        ) {
+            let container = builder.nodes.ancestors(builder.current_node_id).find_map(|node| {
+                if let AstKind::TSConditionalType(ty) = node.kind() {
+                    return ty.extends_type.span().contains_inclusive(self.span).then_some(ty);
+                }
+                None
```

Comment:
It looks like the `infer` only is allowed in `extendType` of `TSConditionalType`, see [TypeScript Playground](It looks like the `infer` only allowed in `extendType` of `TSConditionalType`, so that we can simplify this to check scope flags. 
), so that we can simplify this to check scope flags. 

The code looks like

```rs
    let scope_id = builder
                .scope
                .ancestors(builder.current_scope_id)
                .find_map(|scope_id| builder.scope.get_flags(scope_id).is_ts_conditional())
```

Reply:
mind expanding a bit more?

if we were in consequent of extendType, then viewed scope ancestors, wouldn't we find the `ts_conditional` scope, which is incorrect?

don't we also need to check that this is the `extendType` ?

Reply:
oh ignore me. i understand now.